File: domain-driven-design/node-web/src/utils/logging.js
import fs from 'fs';
import path from 'path';

/**
 * 设置日志文件路径
 * @param {string} logFile - 日志文件的路径
 */
let logFilePath = '';  // 声明一个全局变量，用来存储日志文件路径

export function setupLogging(logFile) {
  try {
    // 使用 import.meta.url 获取当前模块的路径，并推导出项目根目录路径
    const currentDir = path.dirname(new URL(import.meta.url).pathname);
    
    // 获取项目根目录路径（假设你是在根目录的某个模块里设置日志路径）
    const rootDir = path.resolve(currentDir, '../..'); // 返回项目根目录

    // 设置日志文件路径为根目录
    logFilePath = path.join(rootDir, logFile); // 现在日志文件路径在根目录下
    
    // 确保日志文件存在，不存在时创建文件
    fs.appendFileSync(logFilePath, '', 'utf8');
    console.log(`日志系统初始化完成，日志写入: ${logFilePath}`);
  } catch (err) {
    console.error(`无法初始化日志文件 ${logFile}: ${err}`);
    process.exit(1);
  }
}

/**
 * 记录请求日志
 * @param {object} req - 请求对象
 * @param {Date} start - 请求开始时间
 */
export function logRequest(req, start) {
  const duration = Date.now() - start;
  const logMessage = `REQUEST: ${req.method} ${req.url} took ${duration}ms\n`;
  logToFile(logMessage);
}

/**
 * 记录普通信息日志
 * @param {string} message - 要记录的信息
 */
export function logInfo(message) {
  const logMessage = `INFO: ${message}\n`;
  logToFile(logMessage);
}

/**
 * 记录错误日志
 * @param {string} message - 错误信息
 */
export function logError(message) {
  const logMessage = `ERROR: ${message}\n`;
  logToFile(logMessage);
}

/**
 * 将日志消息写入文件
 * @param {string} message - 日志消息
 */
function logToFile(message) {
  try {
    fs.appendFileSync(logFilePath, message, 'utf8');
  } catch (err) {
    console.error(`无法写入日志文件 ${logFilePath}: ${err}`);
  }
}

File: domain-driven-design/python-web/src/utils/logging.py
# -*- coding: utf-8 -*-
from datetime import datetime
import os
import time
from src.config.server_config import LOGGING

# 全局变量，用于存储日志文件路径
log_file_path = None

# 设置日志文件路径
def setup_logging(log_file):
    global log_file_path
    log_file_path = log_file

    # 确保日志文件存在
    if not os.path.exists(log_file_path):
        with open(log_file_path, 'w'): pass  # 创建空文件
    print(f"日志系统初始化完成，日志写入: {log_file_path}")

# 记录中间件日志
def log_middleware(middleware, req, start_time):
    duration = (time.time() - start_time) * 1000  # 转换为毫秒
    formatted_time = datetime.fromtimestamp(start_time).strftime("%Y-%m-%d %H:%M:%S")
    log_message = f"{middleware}: {req.command} {req.path} at {formatted_time} took {duration:.2f}ms\n"
    log_to_file(log_message)

# 记录请求日志
def log_request(req, start_time):
    duration = (time.time() - start_time) * 1000  # 转换为毫秒
    log_message = f"REQUEST: {req.command} {req.path} at {start_time}"
    log_to_file(log_message)

# 记录普通信息日志
def log_info(message):
    log_message = f"INFO: {message}\n"
    log_to_file(log_message)

# 记录错误日志
def log_error(message):
    log_message = f"ERROR: {message}\n"
    log_to_file(log_message)

# 将日志消息写入文件
def log_to_file(message):
    global log_file_path
    print(f'log_to_file message: {message}')
    if log_file_path:
        with open(log_file_path, 'a', encoding='utf-8') as log_file:
            log_file.write(message)
    else:
        print("日志文件路径未初始化，无法写入日志。")
